interface/forms/group_attendance/functions.php

```diff
@@ -0,0 +1,214 @@
+<?php
+
+/**
+ * interface/forms/group_attendance/functions.php functions for form
+ *
+ * Contains the functions used for the group attendance form
+ *
+ * Copyright (C) 2016 Shachar <shachar058@gmail.com>
+ * Copyright (C) 2016 Amiel<amielboim@gmail.com>
+ *
+ *
+ * LICENSE: This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 3
+ * of the License, or (at your option) any later version.
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://opensource.org/licenses/gpl-license.php>;.
+ *
+ * @package OpenEMR
+ * @author  Shachar Zilbershlag <shachar058@gmail.com>
+ * @author  Amiel Elboim <amielboim@gmail.com>
+ * @link    http://www.open-emr.org
+ */
+
+require_once("../../globals.php");
+require_once("$srcdir/api.inc");
+require_once("$srcdir/forms.inc");
+
+/**
+ * Returns form_id of an existing attendance form for group encounter (if one already exists);
+ * @param $encounter
+ * @param $group_id
+ * @return array|null
+ */
+function get_form_id_of_existing_attendance_form($encounter, $group_id){
+    $sql = "SELECT form_id FROM forms WHERE encounter = ? AND form_name = 'Group Attendance Form' AND therapy_group_id = ? AND deleted = 0;";
+    $result = sqlQuery($sql, array($encounter, $group_id));
+    return $result;
+}
+
+/**
+ * Inserts participant data into DB
+ * @param $form_id
+ * @param $therapy_group
+ * @param $group_encounter_data
+ * @param $appt_data
+ */
+function participant_insertions($form_id, $therapy_group, $group_encounter_data,$appt_data){
+    $patientData = $_POST['patientData'];
+    foreach ($patientData as $pid => $patient){
+
+        //Insert into therapy_groups_participants_attendance table
+        insert_into_tgpa_table($form_id, $pid, $patient);
+
+        //Check if to create appt and encounter for each patient (if has certain status and 'bottom' submit was pressed, not 'add_patient' submit).
+        $create_for_patient = if_to_create_for_patient($patient['status']);
+        if($create_for_patient){
+
+            //Create appt for each patient (if there is appointment connected to encounter)
+            if(!empty($appt_data)){
+                insert_patient_appt($pid, $therapy_group, $appt_data['pc_aid'], $appt_data['pc_eventDate'], $appt_data['pc_startTime'], $patient);
+            }
+
+            //Create encounter for each patient
+            insert_patient_encounter($pid, $therapy_group, $group_encounter_data['date'], $patient, $appt_data['pc_aid']);
```

Comment:
Ok now returns pc_eid and encounter id (respectively). Just to be clear - what is fetched from form_encounter is not the `id` value but the`encounter` value (that's what I think you meant). 
Waiting for further instruction.